pkg/agent/cniserver/sriov.go

```diff
@@ -0,0 +1,144 @@
+// +build linux
+
+// Copyright 2021 Antrea Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cniserver
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/containernetworking/cni/pkg/types/current"
+	sriovcniutils "github.com/k8snetworkplumbingwg/sriov-cni/pkg/utils"
+	"google.golang.org/grpc"
+	"k8s.io/klog/v2"
+
+	// Version v1 of the Kubelet API was introduced in K8s v1.20.
+	// Using version v1alpha1 instead to support older K8s versions.
+	podresourcesv1alpha1 "k8s.io/kubelet/pkg/apis/podresources/v1alpha1"
+
+	"antrea.io/antrea/pkg/agent/util"
+)
+
+const (
+	kubeletPodResourcesPath = "/var/lib/kubelet/pod-resources"
+	kubeletSocket           = "kubelet.sock"
+	connectionTimeout       = 10 * time.Second
+)
+
+type KubeletPodResources struct {
+	resources []*podresourcesv1alpha1.PodResources
+}
+
+func GetPodResourceRequests(podName string, podNamespace string) (string, error) {
```

Comment:
I don't think this function is called yet, so there is no usage example, but I'm curious as to why you are returning the device IDs as a space-separated string, instead of as a slice of strings? Wouldn't it make consumption of the result easier?

Reply:
Yes; updated it.